plugins/s3-viewer-backend/src/service/plugin.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  BucketStatsProvider,
+  BucketsProvider,
+  CredentialsProvider,
+  S3Api,
+  s3ViewerExtensionPoint,
+} from '@spreadshirt/backstage-plugin-s3-viewer-node';
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { S3Builder } from './S3Builder';
+import express from 'express';
+
+export const s3ViewerPlugin = createBackendPlugin({
+  pluginId: 's3',
+  register(env) {
+    let s3Client: S3Api;
+    let s3CredentialsProvider: CredentialsProvider;
+    let s3BucketsProvider: BucketsProvider;
+    let s3BucketStatsProvider: BucketStatsProvider;
+    let s3Middleware: express.RequestHandler;
+
+    env.registerExtensionPoint(s3ViewerExtensionPoint, {
+      setClient(client) {
+        if (s3Client !== undefined) {
+          throw new Error('A S3 client has been already set');
+        }
+        s3Client = client;
+      },
+      setCredentialsProvider(credentialsProvider) {
+        if (s3CredentialsProvider !== undefined) {
+          throw new Error('A credentials provider has been already set');
+        }
+        s3CredentialsProvider = credentialsProvider;
+      },
+      setBucketsProvider(bucketsProvider) {
+        if (s3BucketsProvider !== undefined) {
+          throw new Error('A buckets provider has been already set');
+        }
+        s3BucketsProvider = bucketsProvider;
+      },
+      setBucketStatsProvider(bucketStatsProvider) {
+        if (s3BucketStatsProvider !== undefined) {
+          throw new Error('A bucket stats provider has been already set');
+        }
+        s3BucketStatsProvider = bucketStatsProvider;
+      },
+      setMiddleware(middleware) {
+        if (s3Middleware !== undefined) {
+          throw new Error('A middleware has been already set');
+        }
+        s3Middleware = middleware;
+      },
+    });
+
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+        scheduler: coreServices.scheduler,
+        discovery: coreServices.discovery,
+        identity: coreServices.identity,
+        permissions: coreServices.permissions,
+        tokenManager: coreServices.tokenManager,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({
+        logger,
+        config,
+        scheduler,
+        discovery,
+        identity,
+        permissions,
+        tokenManager,
+        httpRouter,
+      }) {
+        let builder = S3Builder.createBuilder({
+          logger,
+          config,
+          scheduler,
+          discovery,
+          identity,
+          permissions,
+          tokenManager,
+        });
```

Comment:
We repeat all of these parameters three times here, could we change the parameter of `init` to just `deps` and then pass that to `createBuilder` and use `deps.httpRouter` below?

Reply:
Simplified the code as suggested